lib/helper/generateBackfillTableHelper.py

```diff
@@ -0,0 +1,138 @@
+import os
+import sys
+import json
+from ..common.logConfig import get_logger
+from lib.common.commonUtil import NetworkUtil
+from lib.common.commonUtil import CommonUtil
+from archiveMozillaHelper import ArchiveMozillaHelper
+from perfherderDataQueryHelper import PerfherderDataQueryHelper
+
+
+logger = get_logger(__name__)
+
+
+class GenerateBackfillTableHelper(object):
+    DEFAULT_HG_QUERY_REVISION_JSON_URL = "https://hg.mozilla.org/mozilla-central/json-rev"
+    DEFAULT_BACKFILL_TABLE_LOCAL_FN = "backfill_table.json"
+
+    @staticmethod
+    def generate_archieve_revision_relation_table(input_history_backfill_table, input_backfill_days=14, input_backfill_repo="mozilla-central", input_app_name="firefox", input_channel_name="nightly"):
+        """
+        generate archive dir and revision relation table
+        @param input_backfill_days:
+        @param input_backfill_channel:
+        @param input_app_name:
+        @param input_channel_name:
+        @return:
+        example:
+        {
+
+        1505812350: {'pkg_json_url': u'https://archive.mozilla.org/pub/firefox/nightly/2017/09/2017-09-19-10-04-05-mozilla-central/firefox-57.0a1.en-US.mac.json',
+                     'pkg_fn_url': u'https://archive.mozilla.org/pub/firefox/nightly/2017/09/2017-09-19-10-04-05-mozilla-central/firefox-57.0a1.en-US.mac.dmg',
+                     'revision': u'e4261f5b96ebfd63e7cb8af3035ff9fea90c74a5'}
+        }
+        """
+        archieve_revision_relation_table = {}
+        backfill_repo_name = input_backfill_repo + "/"
+        history_backfill_url_dict = {}
+        history_backfill_dir_list = []
+
+        # init platform value
+        if sys.platform == "linux2":
```

Comment:
should adding `input_platform` parameter for selecting specify platform, and only using current platform when input parameter is None